core/controllers/editor.py

```diff
@@ -145,10 +154,10 @@ def put(self, exploration_id):
             self.user, exploration_rights)
 
         try:
-            if can_edit:
+            if can_edit and changes_are_mergeable:
```

Comment:
No, I don't think so that we should raise an error when the changes are not mergeable because we are already showing the UI modal directly.
Yea we show a modal in UI if the ```changes_are_mergeable``` is False.